tests/e2e/tests/test_redirects.py

```diff
@@ -83,25 +83,36 @@ def test_that_checks_redirect_using_incorrect_query_values(self, base_url):
             self.response_info_failure_message(base_url, param, response)
 
     @pytest.mark.parametrize('os', _os)
-    @pytest.mark.parametrize('locale', _locales, ids=lambda l: l.lang)
-    def test_verify_locales_redirect_to_the_expected_product(self, base_url, locale, os):
+    def test_verify_locales_redirect_to_the_expected_product(self, base_url, locale, os, version):
         """Verifies the downloaded version of Firefox matches the expected version number
         and filename when Firefox is requested for a specific OS and locale.
-
-        The test verifies the following aliases: firefox-latest, firefox-esr-latest,
-        firefox-nightly-latest, firefox-beta-latest, firefox-aurora-latest.
         """
-        lang = locale.lang
         # Ja locale has a macOS-specific locale
-        if lang == 'ja' and os == 'osx':
-            lang = 'ja-JP-mac'
-
-        for version in locale.versions:
-            get_params = {
-                'product': 'firefox-' + version,
-                'lang': lang,
-                'os': os
-            }
-
-            fx_pkg_name = self.get_expected_fx_pkg_str(os, 'latest', version)
-            self.verify_redirect_to_correct_product(base_url, fx_pkg_name, get_params)
+        if locale == 'ja' and os == 'osx':
+            locale = 'ja-JP-mac'
+
+        get_params = {
+            'product': 'firefox-' + version,
+            'lang': locale,
+            'os': os
+        }
+
+        fx_pkg_name = self.get_expected_fx_pkg_str(os, 'latest', version)
+        self.verify_redirect_to_correct_product(base_url, fx_pkg_name, get_params)
+
+
+def pytest_generate_tests(metafunc):
+    if 'locale' in metafunc.fixturenames:
+        argvalues = []
+
+        def valid_version(version):
+            # there are no builds for aurora locales
+            return 'a' not in version
+
+        locales = utils.get_firefox_locales()
+        _versions = filter(valid_version, utils.get_product_mappings().values())
+        for locale, versions in locales.items():
+            print(locale, versions, _versions)
```

Comment:
What's the purpose of the `print()` statement?

Reply:
D'oh! debug! I'll fix it up.

Reply:
I now bestow you with an Eagle Eye pin, @m8ttyB